src/program/error/syntax.rs

```diff
@@ -0,0 +1,49 @@
+use std::{error::Error, fmt};
+
+use crate::parser::{LexError, ParseError};
+
+/// Failed to parse a Quil program for some reason.
```

Comment:
```suggestion
/// Failed to deserialize a Quil program for some reason.
```

Just to avoid two diverging meanings of parse in the same file